packages/flutter_tools/lib/src/build_system/build_system.dart

```diff
@@ -322,18 +320,18 @@ class Environment {
   ///
   /// [engineVersion] should be set to null for local engine builds.
   factory Environment({
-    @required Directory projectDir,
-    @required Directory outputDir,
-    @required Directory cacheDir,
-    @required Directory flutterRootDir,
-    @required FileSystem fileSystem,
-    @required Logger logger,
-    @required Artifacts artifacts,
-    @required ProcessManager processManager,
-    @required Platform platform,
-    @required String engineVersion,
-    @required bool generateDartPluginRegistry,
-    Directory buildDir,
+    required Directory projectDir,
+    required Directory outputDir,
+    required Directory cacheDir,
+    required Directory flutterRootDir,
+    required FileSystem fileSystem,
+    required Logger logger,
+    required Artifacts artifacts,
+    required ProcessManager processManager,
+    required Platform platform,
+    String? engineVersion,
```

Comment:
`engineVersion`  is "`null` if built with a local engine" and null is handled everywhere, so make it not required.